src/api/index.tsx

```diff
@@ -0,0 +1,64 @@
+import { ApiState } from '../reducers/api';
+
+export enum HttpMethod {
+  DELETE = 'DELETE',
+  GET = 'GET',
+  PATCH = 'PATCH',
+  POST = 'POST',
+  PUT = 'PUT',
+}
+
+type CallApiParams = {
+  endpoint: string;
+  method?: HttpMethod;
+  credentials?: boolean;
+  version?: string;
+  apiState: ApiState;
+};
+
+type Headers = {
+  [name: string]: string;
+};
+
+export const callApi = async ({
+  apiState,
+  endpoint,
+  credentials = false,
+  method = HttpMethod.GET,
+  version = 'v4',
+}: CallApiParams): Promise<object | { error: Error }> => {
+  if (!endpoint.startsWith('/')) {
+    endpoint = `/${endpoint}`;
+  }
+  if (!endpoint.endsWith('/')) {
+    endpoint = `${endpoint}/`;
+  }
+
+  const headers = {} as Headers;
+  if (apiState.authToken) {
+    headers['Authorization'] = `Bearer ${apiState.authToken}`;
+  }
+
+  try {
+    const response = await fetch(`/api/${version}${endpoint}`, {
+      method,
+      headers,
+      credentials: credentials ? 'include' : undefined,
```

Comment:
>     * Default `credentials` to `true` in `callApi()` so that all endpoints get session cookies by default. Most of them will need session cookies anyway.

I can understand why `omit` breaks but I don't think we need to send the cookies all the time for XHR requests.

> Setting credentials: undefined here still lets me log out in Firefox Release. I see the cookies when I log in and see them get deleted when logging out. I think that maybe some polyfill magic is involved here, doing surprising things when credentials is undefined.

I'd like to better understand why this happens, because I really tried to not add `credentials` when implementing the logout function.

Reply:
Okay so I am not sure to understand why everything works without `credentials` actually. I remove everything and login/logout still works, huh. On this basis, I removed all the code around `credentials` since we don't need it.